icu4c/source/common/ucurr.cpp

```diff
@@ -2703,19 +2609,18 @@ ucurr_getNumericCode(const char16_t* currency) {
     if (currency && u_strlen(currency) == ISO_CURRENCY_CODE_LENGTH) {
         UErrorCode status = U_ZERO_ERROR;
 
-        UResourceBundle *bundle = ures_openDirect(nullptr, "currencyNumericCodes", &status);
-        ures_getByKey(bundle, "codeMap", bundle, &status);
+        LocalUResourceBundlePointer bundle(ures_openDirect(nullptr, "currencyNumericCodes", &status));
```

Comment:
This is different from what you're doing in `_findMetaData()` above, there the result from `ures_openDirect()` is stored in a plain pointer upon which `ures_close()` never gets called while the result from the subsequent call to `ures_getByKey()` is stored in a `LocalUResourceBundlePointer`, the opposite of what's being done here. Why are these different, is that intentional, is that correct?